### PR TITLE
[issue-50] add `initialize` for storage, call `get_initial_state` ret…

### PR DESCRIPTION
**Pull request by eliasyaoyc**
PR #50 

Add `initialize` for storage, call `get_initial_state` returns None if not init storage.

### CODE REVIEW

File: openraft/src/storage.rs [hunk 1]
```diff
@@ -111,6 +111,13 @@ where
     /// for details on where and how this is used.
     type SnapshotData: AsyncRead + AsyncWrite + AsyncSeek + Send + Sync + Unpin + 'static;
 
+    /// Initialize the storage(e.g. state).
+    ///
+    /// When the Raft node is first started, it will call this interface on the storage system to
+    /// initialize node related information,e.g. node state then `InitialState::new_initial` should
+    /// be used.
+    async fn initialize(&self) -> Result<(), StorageError>;
```
Comment (drmingdrmer) [inline openraft/src/storage.rs:119]: IMO, a store does not need to provide an API  `initialize()`. 🤔 

When initializing, RaftCore could just do this:
```
let state = self.sto.get_initial_state().await?;
if state.is_none() {
        let state = InitialState::new_initial(self.id);
        self.sto.save_hard_state(&state.hard_state).await?;
}
```

**Review by drmingdrmer — COMMENTED** [review]
Appreciating~

What about totally getting rid of `initialize()`?

File: openraft/src/storage.rs (hunk 1 above)
Comment (eliasyaoyc) [reply openraft/src/storage.rs:119]: okk, revise immediately

File: memstore/src/test.rs [hunk 2]
```diff
@@ -273,7 +273,13 @@ where
...
         };
 
-        let initial = store.get_initial_state().await?;
+        let initial = if let Some(state) = store.get_initial_state().await?{
+            state
+        }else {
+            let state = InitialState::new_initial(NODE_ID);
+            store.save_hard_state(&state.hard_state).await?;
+            state
+        };
```
Comment (drmingdrmer) [inline memstore/src/test.rs:282]: This is just a test for `store.get_initial_state()`.

Just to check the returned value is enough.

Building a new value with `InitialState::new_initial(NODE_ID)` is only required when booting up a RaftCore.

Comment (eliasyaoyc) [reply memstore/src/test.rs:282]: Gotcha

**Review by drmingdrmer — APPROVED** [review]
👍 👍 nice shot~

**Pull request merged**